notebooks/train_notebook.ipynb

```diff
@@ -0,0 +1,1144 @@
+{
+  "cells": [
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "p5a2OcE_GwrT",
+        "outputId": "4904f2e5-374b-4622-9316-4dc9d615aca1"
+      },
+      "outputs": [],
+      "source": [
+        "# !pip install -U \"git+https://github.com/ab7289-tandon-nyu/csgy6953_DeepLearning_Midterm.git\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# # connect to our wandb project\n",
+        "# !pip install wandb\n",
+        "# !wandb login \"API_KEY\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "**Imports**"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {
+        "id": "Ow3lduSvGwrU"
+      },
+      "outputs": [
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "/home/alodie/miniconda3/envs/avalanche-dev-env/lib/python3.8/site-packages/tqdm/auto.py:22: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+            "  from .autonotebook import tqdm as notebook_tqdm\n"
+          ]
+        }
+      ],
+      "source": [
+        "import torch\n",
+        "import torch.nn as nn\n",
+        "import random\n",
+        "import sys \n",
+        "sys.path.append(\"..\")\n",
+        "import time\n",
+        "import gc"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 4,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "NJaOta3cmQwI",
+        "outputId": "81ddc041-d1ee-410d-ab80-e9a75890b761"
+      },
+      "outputs": [],
+      "source": [
+        "# from google.colab import drive\n",
+        "# drive.mount('/content/drive')"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "**Setup**"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 5,
+      "metadata": {
+        "id": "c2CF9rXbGwrU"
+      },
+      "outputs": [],
+      "source": [
+        "SEED = 1234\n",
+        "\n",
+        "random.seed(SEED)\n",
+        "torch.manual_seed(SEED)\n",
+        "torch.cuda.manual_seed(SEED)\n",
+        "torch.backends.cudnn.deterministic = True\n",
+        "\n",
+        "device = torch.device(\"cuda:1\" if torch.cuda.is_available() else \"cpu\")"
```

Comment:
Perhaps we need to change "cuda:1" to "cuda"? When running `model = model.to(decide)` on Google Colab, I got:
* RuntimeError: CUDA error: invalid device ordinal
* CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
* For debugging consider passing CUDA_LAUNCH_BLOCKING=1.

Reply:
I think a comment would be sufficient.

Reply:
fixed cuda device id issue, just will keep it "cuda"